provider/pihole/client.go

```diff
@@ -224,6 +225,10 @@ func (p *piholeClient) apply(ctx context.Context, action string, ep *endpoint.En
 	log.Infof("%s %s IN %s -> %s", action, ep.DNSName, ep.RecordType, ep.Targets[0])
 
 	form := p.newDNSActionForm(action, ep)
+	if strings.Contains(ep.DNSName, "*") {
+		log.Errorf("UNSUPPORTED: Pihole DNS names cannot return wildcard")
+		return provider.SoftError
```

Comment:
```suggestion
		return provider.NewSoftError(errors.New("UNSUPPORTED: Pihole DNS names cannot return wildcard"))
```

SoftErrors are already logged, see https://github.com/kubernetes-sigs/external-dns/pull/4166